pkg/snapshot/controllers/snapshotschedule.go

```diff
@@ -37,92 +41,126 @@ const (
 	SnapshotSchedulePolicyTypeAnnotation = "stork.libopenstorage.org/snapshotSchedulePolicyType"
 )
 
+// NewSnapshotScheduleController creates a new instance of SnapshotScheduleController.
+func NewSnapshotScheduleController(mgr manager.Manager, r record.EventRecorder) *SnapshotScheduleController {
+	return &SnapshotScheduleController{
+		client:   mgr.GetClient(),
+		Recorder: r,
+	}
+}
+
 // SnapshotScheduleController reconciles VolumeSnapshotSchedule objects
 type SnapshotScheduleController struct {
+	client runtimeclient.Client
+
 	Recorder record.EventRecorder
 }
 
 // Init Initialize the snapshot schedule controller
-func (s *SnapshotScheduleController) Init() error {
+func (s *SnapshotScheduleController) Init(mgr manager.Manager) error {
 	err := s.createCRD()
 	if err != nil {
-		return err
+		return fmt.Errorf("register crd: %s", err)
 	}
-	return controller.Register(
-		&schema.GroupVersionKind{
-			Group:   stork.GroupName,
-			Version: stork_api.SchemeGroupVersion.Version,
-			Kind:    reflect.TypeOf(stork_api.VolumeSnapshotSchedule{}).Name(),
-		},
-		"",
-		1*time.Minute,
-		s)
+	mgr.GetCache()
```

Comment:
Why is this required only in this controller?

Reply:
There is no need for this here, removing.